### DISCUSSION TITLE
Prevent phone's back button from going to previous browser page when modal popup is open, NextJS approuter

### DISCUSSION THREAD

**Discussion by TamirCode** [vercel/next.js#65705] [Help]
### Summary

I would like to hijack the browser's (or the phone's) back button behavior when the modal is open, so that it would close the modal instead of going to previous page in the browser, this is the intuitive behavior for mobile users.

I am using typescript, nextjs, with the new approuter. 
and MUI modal for simplicity:
```tsx
"use client"
import { Modal } from '@mui/material'
import { useState } from 'react'

export default function Component() {
    const [open, setOpen] = useState(false)

    return (
        <>
            <Modal
                open={open}
                onClose={() => setOpen(false)}
                style={{ display: "flex", alignItems: "center", justifyContent: "center" }}
            >
                <div style={{ background: "white", padding: "100px" }}>
                    hi
                </div>
            </Modal>
            
            <button onClick={() => setOpen(true)}>
                open modal
            </button>
        </>
    )
}
```
With the new docs on approuter and useRouter, I am very confused and don't know how to achieve this.
Thank you for the help

**Comment by SouthLink** [comment]
Since user regression is browser behavior, all you can do is add [beforeunload event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) to invoke the browser pop-up that prevents page exit by default, and you can't block user regression manually

**Reply by TamirCode** [reply]
i need help using the new approuter, that is vanilla, and i dont think it will help in this situation since if someone decides to go to another page then it will cancel his event and not just cancel back button

**Reply by SouthLink** [reply]
What I mean is that the browser's history object does not allow us to delete or modify the history for security reasons, nor does it allow us to disable the backdown/forward function, it is not possible to intercept the browser backdown event and turn the Modal component off when it is ejected. It doesn't matter if you use approuter and useRouter, because you can't listen for user back events
